tutorials/how_to/training_transformer_elmo.md

```diff
@@ -0,0 +1,79 @@
+# Training Transformer ELMo
+
+This document describes how to train and use a transformer-based version of ELMo with `allennlp`. The model is a port of the the one described in [Dissecting Contextual Word Embeddings: Architecture and Representation](https://www.semanticscholar.org/paper/4dc99343fdc57cf974746e9549c6ee56f013cee5) by Peters et al.
+
+## Training
+
+1. Obtain training data from http://www.statmt.org/lm-benchmark/1-billion-word-language-modeling-benchmark-r13output.tar.gz.
+    ```
+    export BIDIRECTIONAL_LM_DATA_PATH='/your/path/to/1-billion-word-language-modeling-benchmark-r13output'
+    export BIDIRECTIONAL_LM_TRAIN_PATH=$BIDIRECTIONAL_LM_DATA_PATH'/training-monolingual.tokenized.shuffled/*'
+    ```
+2. Obtain vocab.
+    ```
+    mkdir /your/path/to/vocabulary
+    export BIDIRECTIONAL_LM_VOCAB_PATH=/your/path/to/vocabulary
+    cd $BIDIRECTIONAL_LM_VOCAB_PATH
+    aws s3 cp s3://allennlp/models/elmo/vocab-2016-09-10.txt tokens.txt
+    sed -i '' 's/<UNK>/@@UNKNOWN@@/' tokens.txt
```

Comment:
Should be `sed 's/<UNK>/@@UNKNOWN@@/' tokens.txt`? Otherwise I got a `sed: can't read s/<UNK>/@@UNKNOWN@@/: No such file or directory` error message :)

Reply:
Thanks for catching this! We need to modify the file, hence the  `-i` inplace option. The `sed -i ''` was an attempt to be compatible with BSD sed, but apparently it breaks GNU sed. I've changed the instructions to something more vanilla and compatible.